src/pages/messages/components/MessageList.tsx

```diff
@@ -36,26 +35,23 @@ export const TicketMessageList = ({ ticket, refetchData, canDeleteMessage }: Mes
           m: 0,
           gap: '0.25rem',
         }}>
-        {messages.map((message) => (
-          <MessageItem
-            key={message.identifier}
-            text={message.text}
-            date={message.createdDate}
-            username={message.sender}
-            backgroundColor={ticketColor[ticket.type]}
-            menuElement={
-              !!user &&
-              (canDeleteMessage || user.nvaUsername === message.sender) && (
-                <MessageMenu
-                  ticketId={ticket.id}
-                  refetchData={refetchData}
-                  canDeleteMessage={!!message.text}
-                  messageIdentifier={message.identifier}
-                />
-              )
-            }
-          />
-        ))}
+        {messages.map((message) => {
+          const canDeleteMessage = !!user && userCanDeleteMessage(user, message, ticket.type);
+          return (
+            <MessageItem
+              key={message.identifier}
+              text={message.text}
+              date={message.createdDate}
+              username={message.sender}
+              backgroundColor={ticketColor[ticket.type]}
+              menuElement={
+                canDeleteMessage && (
+                  <MessageMenu messageId={message.id} refetchData={refetchData} disableDelete={!!message.text} />
```

Comment:
`<MessageMenu>` vises bare om `canDeleteMessage==true`. Jeg leser det som at det da er fasit, og at man ikke trenger en egen prop for å fortelle om melding kan slettes. Om den ikke kan slettes vil den ikke rendres uansett 🤔 
Mulig det er for å håndtere tilfelle hvor meldingen allerede _er_ slettet, men da burde kanskje `canDeleteMessage` vært `false` uansett?

tl;dr: Virker rart at `canDeleteMessage` og `disableDelete` skal baseres på ulike ting og ikke må samsvare

```suggestion
                canDeleteMessage && (
                  <MessageMenu messageId={message.id} refetchData={refetchData} />
```

Reply:
Ser ut som den er der for å disable slett-knappen til noe som allerede er sletta ja. Da ville jeg nok heller tatt vekk selve meny-knappen. Feks med en `if (!message.text) return false;` i `userCanDeleteMessage()`

![bilde](https://github.com/user-attachments/assets/f83bda8d-1920-4837-9b81-d664116184ac)

Reply:
Husler vi diskuterte dette når denne funksjonaliteten først ble implementert. Tanken bak var at vi ikke skulle fjerne innhold uten "context", og at GUI skulle være likt så lenge det lar seg gjøre. Men på den andre siden hadde jeg akkurat samme tanker som deg når jeg så koden, så tenker det er innafor å ta vekk knappen 👍

Reply:
Godt poeng et også, så opp til deg!